userspace/graph-store/sur-file-walkthrough.md

```diff
@@ -0,0 +1,345 @@
+
+`content` enumerates all the possible content types that a post can have. The possible content types can be:
+
+- **Text** - representing plain text content
+- **Url** - specific data type for urls
+- **Mention** - mentioning another ship
+- **Code** - a pair of a piece of code that was executed and its result (static data, no execution takes place inside of Graph Store)
+- **Reference** - a reference to another post, using the `reference` type
+
+Currently, these are the only content types supported by Graph Store.
+
+### Post
+```
++$  post
+  $:  author=ship
+      =index
+      time-sent=time
+      contents=(list content)
+      hash=(unit hash)
+      =signatures
+  ==
+::
++$  indexed-post  [a=atom p=post]
+```
+
+Post is a fundamental type that represents what we normally think of as a post on social media. Most of the rest of the types are self-explanatory. `hash` is the optional hash of the post, and `signatures` is the (potentially empty) set of `signature`s if the post is cryptographically signed.
+
+An `indexed-post` is a post with an associated index fragment that can be used to validate a post's index with an index fragment that is expected at the end of the index list.
+
+### Hashing (Part 2)
+```
++$  validated-portion
+  $:  parent-hash=(unit hash)
+      author=ship
+      time-sent=time
+      contents=(list content)
+  ==
+::
+```
+
+The parts of a `post` that are actually hashed to obtain a value of type the earlier type `hash`.
+
+
+## Graph Store
+
+Here's `sur/graph-store.hoon`
+
+```
++$  graph         ((mop atom node) gth)
++$  marked-graph  [p=graph q=(unit mark)]
+::
++$  node          [=post children=internal-graph]
++$  graphs        (map resource marked-graph)
+::
++$  internal-graph
+  $~  [%empty ~]
+  $%  [%graph p=graph]
+      [%empty ~]
+  ==
+::
++$  tag-queries   (jug term resource)
+::
+::
++$  network
+  $:  =graphs
+      =tag-queries
+      =update-logs
+      archive=graphs
+      validators=(set mark)
+  ==
+::
++$  update
+  $%  [%0 p=time q=update-0]
+  ==
+::
++$  update-log    ((mop time logged-update) gth)
++$  update-logs   (map resource update-log)
+::
++$  logged-update
+  $%  [%0 p=time q=logged-update-0]
+  ==
+::
++$  logged-update-0
+  $%  [%add-graph =resource =graph mark=(unit mark) overwrite=?]
+      [%add-nodes =resource nodes=(map index node)]
+      [%remove-nodes =resource indices=(set index)]
+      [%add-signatures =uid =signatures]
+      [%remove-signatures =uid =signatures]
+  ==
+::
+
+
++$  update-0
+  $%  logged-update-0
+      [%remove-graph =resource]
+    ::
+      [%add-tag =term =resource]
+      [%remove-tag =term =resource]
+    ::
+      [%archive-graph =resource]
+      [%unarchive-graph =resource]
+      [%run-updates =resource =update-log]
+    ::
+    ::  NOTE: cannot be sent as pokes
+    ::
+      [%keys =resources]
+      [%tags tags=(set term)]
+      [%tag-queries =tag-queries]
+  ==
+--
```

Comment:
Just realized, this section is out of data based on the latest breaking changes as well. I have to figure out what the new stuff is.